### PR TITLE
[Community Pipeline] Add gluegen

### LINKED ISSUE #6347: Add GlueGen
### Model/Pipeline/Scheduler description

GlueGen is a minimal adapter that allow alignment between any encoder (Text Encoder of different language, Multilingual Roberta, AudioClip) and CLIP text encoder used in standard Stable Diffusion model.

Below are the results produced by using community available weights in English and French (using their French adapter):
**Checkpoint: Anime pencil diffusion**
00004-Gollum du Seigneur des Anneaux style de concept de crayon anime
French Prompt: Gollum du Seigneur des Anneaux style de concept de crayon anime 

00000Gollum from lord of the rings anime pencil concept style
English Prompt: Gollum from lord of the rings anime pencil concept style

**Checkpoint: animeTWO**
00030-une fille heureuse
French Prompt: une fille heureuse
00024a girl happy
English Prompt: a girl happy

I would like to contribute this model into Diffuser library

### Open source status

- [X] The model implementation is available.
- [X] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

From: Salesforce
Code and pretrained weights: https://github.com/salesforce/GlueGen
Paper: https://arxiv.org/pdf/2303.10056.pdf

### PR DESCRIPTION
**Pull request by rootonchair**
# What does this PR do?

Fixes #6347 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by rootonchair** [comment]
Draft test code
```python
from PIL import Image

import torch

from transformers import AutoModel, AutoTokenizer

from examples.community.gluegen import GlueGenStableDiffusionPipeline
from examples.community.gluegen import Translator_noln

if __name__ == "__main__":
    device = "cuda"

    lm_model_id = "xlm-roberta-large"
    token_max_length = 77

    text_encoder = AutoModel.from_pretrained(lm_model_id)
    tokenizer = AutoTokenizer.from_pretrained(lm_model_id, model_max_length=token_max_length, use_fast=False)

    language_adapter = Translator_noln(num_tok=77, dim=1024, dim_out=768)
    language_adapter.load_state_dict(torch.load("gluenet_French_clip_overnorm_over3_noln.ckpt"))
    language_adapter.to(device)

    tensor_norm = torch.Tensor([[43.8203],[28.3668],[27.9345],[28.0084],[28.2958],[28.2576],[28.3373],[28.2695],[28.4097],[28.2790],[28.2825],[28.2807],[28.2775],[28.2708],[28.2682],[28.2624],[28.2589],[28.2611],[28.2616],[28.2639],[28.2613],[28.2566],[28.2615],[28.2665],[28.2799],[28.2885],[28.2852],[28.2863],[28.2780],[28.2818],[28.2764],[28.2532],[28.2412],[28.2336],[28.2514],[28.2734],[28.2763],[28.2977],[28.2971],[28.2948],[28.2818],[28.2676],[28.2831],[28.2890],[28.2979],[28.2999],[28.3117],[28.3363],[28.3554],[28.3626],[28.3589],[28.3597],[28.3543],[28.3660],[28.3731],[28.3717],[28.3812],[28.3753],[28.3810],[28.3777],[28.3693],[28.3713],[28.3670],[28.3691],[28.3679],[28.3624],[28.3703],[28.3703],[28.3720],[28.3594],[28.3576],[28.3562],[28.3438],[28.3376],[28.3389],[28.3433],[28.3191]]).to(device)

    pipeline = GlueGenStableDiffusionPipeline.from_pretrained(
        "runwayml/stable-diffusion-v1-5",
        text_encoder=text_encoder,
        tokenizer=tokenizer,
        language_adapter=language_adapter,
        tensor_norm=tensor_norm,
    ).to(device)

    prompt = "une voiture sur la plage" #+ '__' + "French"

    generator = torch.Generator(device='cuda').manual_seed(42) 
    image = pipeline(prompt, generator=generator).images[0]
    image.save("gluegen_output_fr.png")
```

File: examples/community/gluegen.py [hunk 1]
```diff
@@ -0,0 +1,865 @@
...
+    scale_lora_layers,
+    unscale_lora_layers,
+)
+from diffusers.utils.torch_utils import randn_tensor
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class translator_base(nn.Module):
```
Comment (DN6) [inline examples/community/gluegen.py:28]: Can we use camel case for the class names?

File: examples/community/gluegen.py [hunk 2]
```diff
@@ -0,0 +1,865 @@
...
+        else:
+            x = self.net_sen(x)
+            x = x.transpose(1, 2)
+
+            x = self.net_tok(x)
+            x = x.transpose(1, 2)
+        return x
+
+
+class Translator_noln(nn.Module):
```
Comment (DN6) [inline examples/community/gluegen.py:120]: Can we use camel case for  the class names

**Comment by DN6** [comment]
@rootonchair Requested some small changes regarding class names.

File: examples/community/gluegen.py (hunk 1 above)
Comment (rootonchair) [reply examples/community/gluegen.py:28]: Sure. I will rename the classes to camel case

**Review by DN6 — APPROVED** [review]
Nicely done! 👍🏽
